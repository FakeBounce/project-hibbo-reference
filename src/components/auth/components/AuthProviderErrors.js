import React from 'react';
import { string, number, shape } from 'prop-types';

import Error from './Error';

const AuthProviderErrors = ({
  fbError,
  googleError,
  fbPosition,
  googlePosition,
}) => {
  if (fbError !== '') {
    return (
      <Error
        label={fbError}
        posY={fbPosition.y - 55}
        posX={fbPosition.x + 50}
      />
    );
  } else if (googleError !== '') {
    return (
      <Error
        label={googleError}
        posY={googlePosition.y - 55}
        posX={googlePosition.x + 50}
      />
    );
  }

  return null;
};

AuthProviderErrors.propTypes = {
  fbError: string.isRequired,
  googleError: string.isRequired,
  fbPosition: shape({
    x: number.isRequired,
    y: number.isRequired,
  }).isRequired,
  googlePosition: shape({
    x: number.isRequired,
    y: number.isRequired,
  }).isRequired,
};

export default AuthProviderErrors;
