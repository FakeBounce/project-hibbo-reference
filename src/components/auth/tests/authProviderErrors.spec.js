import React from 'react';
import renderer from 'react-test-renderer';

import AuthProviderErrors from '../components/AuthProviderErrors';

describe('AuthProviderErrors', () => {
  const position = {
    x: 0,
    y: 0,
  };

  it('should render the component', () => {
    const component = renderer
      .create(
        <AuthProviderErrors
          fbError=""
          googleError=""
          fbPosition={position}
          googlePosition={position}
        />,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should render the component on fb error', () => {
    const component = renderer
      .create(
        <AuthProviderErrors
          fbError="Error"
          googleError=""
          fbPosition={position}
          googlePosition={position}
        />,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should render the component on google error', () => {
    const component = renderer
      .create(
        <AuthProviderErrors
          fbError=""
          googleError="Error"
          fbPosition={position}
          googlePosition={position}
        />,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
