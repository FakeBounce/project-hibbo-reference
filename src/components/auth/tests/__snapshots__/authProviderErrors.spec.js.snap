// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthProviderErrors should render the component 1`] = `null`;

exports[`AuthProviderErrors should render the component on fb error 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "left": 50,
        "top": -55,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#FF7171",
        },
      ]
    }
  >
    <View
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Image
          source={1}
          style={undefined}
        />
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          Error
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
          Object {
            "borderBottomColor": "#FF7171",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`AuthProviderErrors should render the component on google error 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "left": 50,
        "top": -55,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#FF7171",
        },
      ]
    }
  >
    <View
      style={undefined}
    >
      <View
        style={undefined}
      >
        <Image
          source={1}
          style={undefined}
        />
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          Error
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
          Object {
            "borderBottomColor": "#FF7171",
          },
        ]
      }
    />
  </View>
</View>
`;
